lib/ansible/modules/system/ipset.py

```diff
@@ -0,0 +1,180 @@
+                self.module.fail_json(msg='Failed to add %s to hash %s: %s' % (key, self.ipset_name, out + err))
+
+        return rc
+
+    def ipset_delete_address(self):
+        rc = ''
+
+        list_keys = self.keys.split(' ')
+        for key in list_keys:
+            cmd = "%s del %s %s" % (self.ipset_cmd, self.ipset_name, key)
+
+            rc, out, err = self.module.run_command(cmd)
+            if rc != 0:
+                self.module.fail_json(msg='Failed to delete %s to hash %s: %s' % (key, self.ipset_name, out + err))
+
+        return rc
+
+    def ipset_create_hashset(self):
+        cmd = "%s create %s %s" % (self.ipset_cmd, self.ipset_name, self.hash_type)
+
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to create ipset %s: %s' % (self.ipset_name, out + err))
+        else:
+            return rc
+
+    def ipset_destroy_hashset(self):
+        cmd = "%s destroy %s" % (self.ipset_cmd, self.ipset_name)
+
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to destroy ipset %s: %s' % (self.ipset_name, out + err))
+        else:
+            return rc
+
+    def configure_ipset(self):
+        rc = False
+
+        if self.state == 'create':
+            rc = self.ipset_create_hashset()
+        elif self.state == 'destroy':
+            rc = self.ipset_destroy_hashset()
+        elif self.state == 'add':
+            rc = self.ipset_add_address()
+        elif self.state == 'delete':
+            rc = self.ipset_delete_address()
+
+        return rc
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            ipset_name=dict(required=True, type='str'),
+            hash_type=dict(choices=['hash:net', 'hash:ip', 'hash:mac', 'hash:ip,mac', 'hash:net,net',
+                                    'hash:ip,port', 'hash:net,port', 'hash:ip,port,ip', 'hash:ip,port,net',
+                                    'hash:ip,mark', 'hash:net,port,net', 'hash:net,iface'],
+                           required=False, default=None, type='str'),
+            keys=dict(required=False, default=None, type='str'),
+            state=dict(choices=['add', 'delete', 'destroy', 'create'], required=True),
+        ),
+        supports_check_mode=True
+    )
+
+    ipset_name = module.params['ipset_name']
+    hash_type = module.params['hash_type']
+    keys = module.params['keys']
+    state = module.params['state']
+
+    if ipset_name is None:
+        module.fail_json(msg='ipset_name cannot be None')
+
+    if state == 'create' and hash_type is None:
+        module.fail_json(msg='hash_type cannot be None when state is create')
+
+    if keys is None:
+        if state == 'add' or state == 'delete':
+            module.fail_json(msg='Keys cannot None when state is add or delete')
+
+    ipset = IpsetModule(module)
+    result = ipset.configure_ipset()
+
+    module.exit_json(changed=result)
```

Comment:
`result` appears to be just a return code, is there any desire to have a more verbose message to provide context to the user about what took place?